test/uri/uri_relative_to_test.cc

```diff
@@ -0,0 +1,72 @@
+#include <gtest/gtest.h>
+#include <sourcemeta/jsontoolkit/uri.h>
+
+TEST(URI_relative_to, absolute_absolute_base_true_1) {
+  const sourcemeta::jsontoolkit::URI base{"https://www.example.com"};
+  sourcemeta::jsontoolkit::URI uri{"https://www.example.com/foo"};
+  uri.relative_to(base);
+  EXPECT_EQ(uri.recompose(), "foo");
+}
+
+TEST(URI_relative_to, absolute_absolute_base_true_2) {
+  const sourcemeta::jsontoolkit::URI base{"https://www.example.com/foo"};
+  sourcemeta::jsontoolkit::URI uri{"https://www.example.com/foo"};
+  uri.relative_to(base);
+  EXPECT_EQ(uri.recompose(), "");
+}
+
+TEST(URI_relative_to, absolute_absolute_base_true_3) {
+  const sourcemeta::jsontoolkit::URI base{"https://www.example.com/foo"};
+  sourcemeta::jsontoolkit::URI uri{"https://www.example.com/foo/bar?q=1"};
+  uri.relative_to(base);
+  EXPECT_EQ(uri.recompose(), "bar?q=1");
+}
+
+TEST(URI_relative_to, absolute_absolute_base_true_4) {
+  const sourcemeta::jsontoolkit::URI base{"https://www.example.com/foo"};
+  sourcemeta::jsontoolkit::URI uri{"https://www.example.com/foo/bar#baz"};
+  uri.relative_to(base);
+  EXPECT_EQ(uri.recompose(), "bar#baz");
+}
+
+TEST(URI_relative_to, absolute_absolute_base_false_1) {
+  const sourcemeta::jsontoolkit::URI base{"https://www.example.com/foo"};
+  sourcemeta::jsontoolkit::URI uri{"http://www.example.com/foo"};
+  uri.relative_to(base);
+  EXPECT_EQ(uri.recompose(), "http://www.example.com/foo");
+}
+
+TEST(URI_relative_to, absolute_absolute_base_false_2) {
+  const sourcemeta::jsontoolkit::URI base{"https://www.example.com/foo"};
+  sourcemeta::jsontoolkit::URI uri{"https://www.example.com"};
+  uri.relative_to(base);
+  EXPECT_EQ(uri.recompose(), "https://www.example.com");
+}
+
+TEST(URI_relative_to, absolute_absolute_base_false_3) {
+  const sourcemeta::jsontoolkit::URI base{"https://www.example.com/foo/bar"};
+  sourcemeta::jsontoolkit::URI uri{"https://www.example.com/foo"};
+  uri.relative_to(base);
+  EXPECT_EQ(uri.recompose(), "https://www.example.com/foo");
+}
+
+TEST(URI_relative_to, absolute_relative_1) {
+  const sourcemeta::jsontoolkit::URI base{"https://www.example.com"};
+  sourcemeta::jsontoolkit::URI uri{"foo"};
+  uri.relative_to(base);
+  EXPECT_EQ(uri.recompose(), "foo");
+}
+
+TEST(URI_relative_to, relative_absolute_1) {
+  const sourcemeta::jsontoolkit::URI base{"foo/bar"};
+  sourcemeta::jsontoolkit::URI uri{"https://www.example.com/foo"};
+  uri.relative_to(base);
+  EXPECT_EQ(uri.recompose(), "https://www.example.com/foo");
+}
+
```

Comment:
I think We should have better name than this :)

Reply:
What do you mean?

Reply:
Like explicitly mentioning what we are testing ‘relative_schemeless_uri’ or whatever make sense :)

Reply:
Ah, yeah, fair enough. We should also do a lot more testing once we implement this purely ourselves. I wonder what edge cases `uriparser` is already sorting out here that we are not aware of